analyzers/src/SonarAnalyzer.CSharp/Rules/ParameterValidationInMethodWalker.cs

```diff
@@ -44,36 +44,39 @@ internal class ParameterValidationInMethodWalker : SafeCSharpSyntaxWalker
 
         public IEnumerable<Location> ArgumentExceptionLocations => argumentExceptionLocations;
 
-        public ParameterValidationInMethodWalker(SemanticModel semanticModel)
-        {
-            this.semanticModel = semanticModel;
-        }
+        public ParameterValidationInMethodWalker(SemanticModel semanticModel) => this.semanticModel = semanticModel;
 
         public override void Visit(SyntaxNode node)
         {
-            if (this.keepWalking &&
-                !node.IsAnyKind(SubMethodEquivalents))  // Don't explore deeper if this node is equivalent to a method declaration
+            if (keepWalking
+                && !node.IsAnyKind(SubMethodEquivalents))  // Don't explore deeper if this node is equivalent to a method declaration
             {
                 base.Visit(node);
             }
         }
 
-        public override void VisitAwaitExpression(AwaitExpressionSyntax node)
-        {
-            this.keepWalking = false;
-        }
+        public override void VisitAwaitExpression(AwaitExpressionSyntax node) => keepWalking = false;
 
         public override void VisitThrowStatement(ThrowStatementSyntax node)
         {
             // When throw is like `throw new XXX` where XXX derives from ArgumentException, save location
-            if (node.Expression is ObjectCreationExpressionSyntax oces &&
-                this.semanticModel.GetSymbolInfo(oces.Type).Symbol?.OriginalDefinition is ITypeSymbol typeSymbol &&
-                typeSymbol.DerivesFrom(KnownType.System_ArgumentException))
+            if (node.Expression is ObjectCreationExpressionSyntax oces
+                && semanticModel.GetSymbolInfo(oces.Type).Symbol?.OriginalDefinition is ITypeSymbol typeSymbol
```

Comment:
This doesn't work as expected in a cases like this:
```cs
var ex = new ArgumentException();
throw ex;

throw GetArgumentExpression();
```
You should use `semanticModel.GetTypeInfo(node.Expression)` instead and add some test cases.

Reply:
this is a good catch! 

and TBH I expect that this is happening quite often across our codebase and it may be an interesting point to improve across all our rules